pkg/brokerapi/brokerapi.go

```diff
@@ -69,14 +75,28 @@ func (b BrokerAPI) GetInstance(ctx context.Context, instanceID string) (domain.G
 	logger := requestScopedLogger(ctx, b.logger).WithData(lager.Data{"instance-id": instanceID})
 	logger.Info("get-instance")
 
-	return b.broker.GetInstance(ctx, instanceID)
+	res, err := b.broker.GetInstance(ctx, instanceID)
+	return res, toApiResponseError(ctx, err)
 }
 
 // Update modifies an existing service instance
 //  PATCH /v2/service_instances/{instance_id}
 func (b BrokerAPI) Update(ctx context.Context, instanceID string, details domain.UpdateDetails, asyncAllowed bool) (domain.UpdateServiceSpec, error) {
-	spec := domain.UpdateServiceSpec{}
-	return spec, errors.New("not implemented")
+	logger := requestScopedLogger(ctx, b.logger).WithData(lager.Data{"instance-id": instanceID})
+	logger.Info("update-service-instance", lager.Data{"plan-id": details.PlanID, "service-id": details.ServiceID})
+
+	res, err := b.broker.Update(ctx, instanceID, details.ServiceID, details.PlanID)
```

Comment:
we could also use the `details.PreviousValues.PlanID` here, to save a plan lookup?